### PR TITLE
feat(pop-drink): add testnet runtime environment

### PR DESCRIPTION
**Pull request by chungquantin**

### CODE REVIEW

**Review by Daanvdplas — APPROVED** [review]
LGTM!

File: crates/pop-drink/src/lib.rs [hunk 1]
```diff
@@ -48,6 +48,39 @@ pub mod devnet {
...
+			pub type Error = crate::error::Error<v0::Error, pop_runtime_testnet::RuntimeError, 3>;
+		}
+	}
+
+	// Types used in the pop runtime.
+	pub type Balance = BalanceFor<Runtime>;
+	pub type AccountId = AccountIdFor<Runtime>;
+
+	/// Converts an AccountId from Pop's runtime to the account ID used in the contract environment.
+	pub fn account_id_from_slice(s: &AccountId) -> pop_api::primitives::AccountId {
```
Comment (evilrobot-01) [inline crates/pop-drink/src/lib.rs:78]: Why is this duplicated? Why not define it in the root and then just export it from each runtime module?

File: crates/pop-drink/src/lib.rs [hunk 2]
```diff
@@ -48,6 +48,39 @@ pub mod devnet {
...
+	use super::*;
+	pub use crate::error::*;
+
+	/// Error related utilities for smart contracts using pop api.
+	pub mod error {
+		pub use pop_runtime_testnet::RuntimeError::*;
+
+		pub use crate::error::*;
+
+		pub mod v0 {
```
Comment (evilrobot-01) [inline crates/pop-drink/src/lib.rs:65]: Missing doc comments - please address all clippy warnings in PR.

**Review by evilrobot-01 — CHANGES_REQUESTED** [review]
No need for the duplicate function when it can be refactored in less than a minute.

Also various clippy warnings about missing doc comments, which should be added due to the nature of this crate - enabling devs to test via a runtime. Everyting public should be documented so that they are easily able to understand what is what when looking at the crate.

File: crates/pop-drink/src/lib.rs (hunk 1 above)
Comment (chungquantin) [reply crates/pop-drink/src/lib.rs:78]: `AccountId` is defined with `Runtime`

Comment (chungquantin) [reply crates/pop-drink/src/lib.rs]: Refactored to macro to reduce code complexity.

**Review by al3mart — APPROVED** [review]
I see all the feedback has been addressed.

Seeing that we are defining `Error` in both modules, `devnet` and `testnet` we could maybe move all together into the macro.

Here you can see how it would look like if everything were moved into the macro - https://github.com/r0gue-io/pop-drink/compare/chungquantin/feat-testnet...r0gue-io:pop-drink:al3mart/error-in-macro?expand=1
(Though I haven't run any tests)

I'd argue that declaring the modules outside of the macro is nicer to read :)

**Comment by chungquantin** [comment]
> I see all the feedback has been addressed.
> 
> Seeing that we are defining `Error` in both modules, `devnet` and `testnet` we could maybe move all together into the macro.
> 
> Here you can see how it would look like if everything were moved into the macro - https://github.com/r0gue-io/pop-drink/compare/chungquantin/feat-testnet...r0gue-io:pop-drink:al3mart/error-in-macro?expand=1 (Though I haven't run any tests)
> 
> I'd argue that declaring the modules outside of the macro makes it nicer to read :)

Reason why I don't put everything inside the macro because I feel like there will be some changes made related to specific runtimes that will break the macro in the future. But can add it back if other reviewers are happy with it.

**Pull request merged**